bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java

```diff
@@ -98,9 +98,8 @@ public void handleCommand(ChannelUID channelUID, Command command) {
      * Stops the API request or websocket reconnect timer
      */
     private void stopTimer() {
-        ScheduledFuture<?> future = scheduledFuture;
```

Comment:
Why do you have away from the local variable concept here? I personally think it makes sense

Reply:
No specific one. I personally do not like this pattern. It only omits the warning in Eclipse IDE. Or did I miss something. But I do not have a problem with reverting it if you want me to do that.

Reply:
From my understanding this keeps the reference even if some other method modifies the field (or sets it to null), so we make sure that we always act on the object that was valid when we assigned the local variable.

Reply:
Good point. Does this mean we will watch out for it and tell people to use it in future PRs?

Reply:
That's right! There's a small possibility the variable is just reassigned after doing the null check. So it's never safe to reference fields unless access is synchronized. So the pattern is there to fix bugs and not to please the IDE. :wink:

Reply:
The pattern is also advocated by the Eclipse null annotations [documentation (The case of fields)](https://help.eclipse.org/oxygen/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_null_annotations.htm):

> The situation is more delicate when considering a field marked `@Nullable`. Such a field should always be considered as dangerous and the recommended way of working with nullable fields is: always assign the value to a local variable before working with it. Using a local variable the flow analysis can tell exactly, whether a dereference is sufficiently protected by a null check. When following this general rule working with nullable fields poses no problems.

Reply:
Thanks. I reverted it.